swift/ql/src/queries/Security/CWE-312/CleartextLogging.ql

```diff
@@ -18,7 +18,9 @@ import codeql.swift.dataflow.DataFlow
 import codeql.swift.security.CleartextLoggingQuery
 import CleartextLoggingFlow::PathGraph
 
-from CleartextLoggingFlow::PathNode src, CleartextLoggingFlow::PathNode sink
-where CleartextLoggingFlow::flowPath(src, sink)
-select sink.getNode(), src, sink, "This $@ is written to a log file.", src.getNode(),
-  "potentially sensitive information"
+from CleartextLoggingFlow::PathNode source, CleartextLoggingFlow::PathNode sink
+where CleartextLoggingFlow::flowPath(source, sink)
+select sink, source, sink,
+  "This operation writes '" + sink.toString() +
+    "' to a log file. It may contain unencrypted sensitive data from $@.", source,
+  source.getNode().toString()
```

Comment:
We have a lot of very different alert messages across languages for this query:
- [Go](https://github.com/github/codeql/blob/main/go/ql/src/Security/CWE-312/CleartextLogging.ql): "$@ flows to a logging call."
- [Python](https://github.com/github/codeql/blob/main/python/ql/src/Security/CWE-312/CleartextLogging.ql): "This expression logs $@ as clear text."
- [JS](https://github.com/github/codeql/blob/main/javascript/ql/src/Security/CWE-312/CleartextLogging.ql): "This logs sensitive data returned by $@ as clear text."
- [Ruby](https://github.com/github/codeql/blob/main/ruby/ql/src/queries/security/cwe-312/CleartextLogging.ql): "This logs sensitive data returned by $@ as clear text."

Are any of those alert messages better than the one we have right now for Swift?

Reply:
The main thing I care about is describing the source in the string, so that in a list of results (e.g. on MRVA) ones that are different from the others immediately stand out (without having to click through).  The phrasing I've gone for here is just similar to the other `swift/cleartext-*` queries.

I do like the Go one for it's simplicity, but I think what we have is clear and consistent.